File: example/cplusplus/a_double_unary_op_xam.cpp
// SPDX-License-Identifier: GPL-3.0-or-later
// SPDX-FileCopyrightText: Bradley M. Bell <bradbell@seanet.com>
// SPDX-FileContributor: 2017-23 Bradley M. Bell
// ----------------------------------------------------------------------------
// a_double unary operators
// -----------------------------------------------------------------------------
// BEGIN SOURCE
# include <cstdio>
# include <cppad/py/cppad_py.hpp>

bool a_double_unary_op_xam(void) {
   using cppad_py::a_double;
   //
   // initialize return variable
   bool ok = true;
   //------------------------------------------------------------------------
   a_double a2(2.0);
   a_double aplus2 = + a2 ;
   a_double aminus2 = - a2 ;
   //
   ok = ok && aplus2 == 2.0;
   ok = ok && aminus2 == -2.0;
   //
   return( ok );
}
// END SOURCE
//
/*
{xrst_begin a_double_unary_op_xam.cpp}

C++: a_double Unary Plus and Minus: Example and Test
####################################################
{xrst_literal
   // BEGIN SOURCE
   // END SOURCE
}
{xrst_end a_double_unary_op_xam.cpp}
*/
//

File: example/python/core/a_double_unary_op_xam.py
# SPDX-License-Identifier: GPL-3.0-or-later
# SPDX-FileCopyrightText: Bradley M. Bell <bradbell@seanet.com>
# SPDX-FileContributor: 2017-22 Bradley M. Bell
# ----------------------------------------------------------------------------
# a_double unary operators
# -----------------------------------------------------------------------------
# BEGIN SOURCE
def a_double_unary_op_xam() :
   #
   import numpy
   import cppad_py
   #
   # initialize return variable
   ok = True
   # ---------------------------------------------------------------------
   a2 = cppad_py.a_double(2.0)
   aplus2 = + a2
   a2_minus = - a2
   #
   ok = ok and aplus2 == 2.0
   ok = ok and a2_minus == -2.0
   #
   return( ok )
#
# END SOURCE
#
#
# {xrst_begin a_double_unary_op_xam_py}
# {xrst_comment_ch #}
#
# Python: a_double Unary Plus and Minus: Example and Test
# #######################################################
# {xrst_literal
#  # BEGIN SOURCE
#  # END SOURCE
# }
# {xrst_end a_double_unary_op_xam_py}
#
